server/src/test/java/org/candlepin/dto/api/v1/SubscriptionTranslatorTest.java

```diff
@@ -0,0 +1,169 @@
+        source.setEndDate(new Date());
+        source.setContractNumber("test_contact");
+        source.setAccountNumber("test_acc_num");
+        source.setModified(new Date());
+        source.setOrderNumber("test_order_num");
+        source.setUpstreamPoolId("test_pool_id");
+        source.setUpstreamEntitlementId("test_ent_id");
+        source.setUpstreamConsumerId("test_cons_id");
+        source.setCertificate(this.createCert());
+        source.setCdn(this.createCdn());
+
+        return source;
+    }
+
+    private Collection<ProductData> createProducts() {
+        return Arrays.asList(
+            new ProductData(),
+            new ProductData()
+        );
+    }
+
+    private ProductData createProduct() {
+        return new ProductData();
+    }
+
+    @Override
+    protected SubscriptionDTO initDestinationObject() {
+        // Nothing fancy to do here.
+        return new SubscriptionDTO();
+    }
+
+    @Override
+    protected void verifyOutput(
+        Subscription source, SubscriptionDTO dto, boolean childrenGenerated) {
+        if (source != null) {
+            assertEquals(source.getId(), dto.getId());
+            assertEquals(source.getQuantity(), dto.getQuantity());
+            assertEquals(Util.toDateTime(source.getStartDate()), dto.getStartDate());
+            assertEquals(Util.toDateTime(source.getEndDate()), dto.getEndDate());
+            assertEquals(source.getContractNumber(), dto.getContractNumber());
+            assertEquals(source.getAccountNumber(), dto.getAccountNumber());
+            assertEquals(Util.toDateTime(source.getModified()), dto.getModified());
+            assertEquals(source.getOrderNumber(), dto.getOrderNumber());
+            assertEquals(source.getUpstreamPoolId(), dto.getUpstreamPoolId());
+            assertEquals(source.getUpstreamEntitlementId(), dto.getUpstreamEntitlementId());
+            assertEquals(source.getUpstreamConsumerId(), dto.getUpstreamConsumerId());
+
+            if (childrenGenerated) {
+                assertNotNull(source.getOwner());
+                assertNotNull(source.getProduct());
+                assertNotNull(source.getDerivedProduct());
+                assertNotNull(source.getCertificate());
+                assertNotNull(source.getCdn());
+
+                for (ProductData product : source.getProvidedProducts()) {
+                    assertNotNull(product);
+                }
+
+                for (ProductData product : source.getDerivedProvidedProducts()) {
+                    assertNotNull(product);
+                }
+            }
+            else {
+
+                assertTrue(isEmpty(dto.getProvidedProducts()));
+                assertTrue(isEmpty(dto.getDerivedProvidedProducts()));
+                assertNull(dto.getOwner());
+                assertNull(dto.getProduct());
+                assertNull(dto.getDerivedProduct());
+                assertNull(dto.getCert());
+                assertNull(dto.getCdn());
+            }
```

Comment:
The various `verifyOutput` methods from the *TranslatorTest.java instances can be used here instead

Reply:
Done.